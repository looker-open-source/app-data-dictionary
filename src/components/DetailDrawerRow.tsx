/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import { Code, TableRow, TableDataCell } from '@looker/components'

import type { ILookmlModelExploreField } from '@looker/sdk'
import type { ColumnDescriptor } from './interfaces'

export const DetailDrawerRow: React.FC<{
  column: ColumnDescriptor
  field: ILookmlModelExploreField
}> = ({ column, field }) => {
  const unformattedValue =
    field[column.rowValueDescriptor as keyof ILookmlModelExploreField]
  const displayValue = unformattedValue ? String(unformattedValue) : undefined
  const Formatter = column.Formatter
  return (
    <TableRow key={column.rowValueDescriptor}>
      <TableDataCell color="text" p="medium" pl="small" pr="small">
        {column.label}
      </TableDataCell>
      <TableDataCell
        className="break"
        color="text3"
        p="medium"
        pl="small"
        pr="small"
      >
        <Code color="text3" fontSize="small" className="break">
          <Formatter
            x={displayValue}
            isRow={false}
            field={field}
            tags={unformattedValue as string[]}
          />
        </Code>
      </TableDataCell>
    </TableRow>
  )
}
