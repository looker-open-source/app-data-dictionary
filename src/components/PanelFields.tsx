/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { useState } from 'react'
import {
  Box,
  ButtonOutline,
  Flex,
  FlexItem,
  Heading,
  InputSearch,
  Paragraph,
  Spinner,
  theme,
} from '@looker/components'
import styled from 'styled-components'
import groupBy from 'lodash/groupBy'
import values from 'lodash/values'
import flatten from 'lodash/flatten'
import toPairs from 'lodash/toPairs'
import orderBy from 'lodash/orderBy'
import type {
  ILookmlModel,
  ILookmlModelExplore,
  ILookmlModelExploreField,
  IUser,
} from '@looker/sdk'
import humanize from 'humanize-string'
import { exploreURL } from '../utils/urls'
import { ExternalLink } from './ExternalLink'
import type { ColumnDescriptor, CommentPermissions } from './interfaces'
import { QuickSearch } from './QuickSearch'
import { ViewOptions } from './ViewOptions'
import { Fields } from './Fields'
import { DIMENSION, MEASURE } from './CategorizedLabel'

export const Main = styled(Box as any)`
  position: relative;
  width: 100%;
  min-height: 93vh;
`

const FullPage = styled(Box as any)`
  position: relative;
  display: flex;
  align-items: stretch;
  justify-content: center;
  align-items: center;
  width: 100%;
  min-height: 93vh;
  flex-direction: column;
`

const IntroText = styled(Paragraph as any)`
  text-align: center;
  margin-top: 5em;
  max-width: 40%;
  color: ${theme.colors.text1};
`

export const ExploreSearch = styled(InputSearch as any)`
  margin-top: 0;
`

export const defaultShowColumns = [
  'label_short',
  'description',
  'name',
  'type',
  'sql',
]

export const PanelFields: React.FC<{
  columns: ColumnDescriptor[]
  currentExplore: ILookmlModelExplore | null
  currentModel: ILookmlModel | null
  loadingExplore: string
  comments: string
  addComment: (newCommentStr: string, field: string) => void
  editComment: (newCommentStr: string, field: string) => void
  deleteComment: (newCommentStr: string, field: string) => void
  authors: IUser[]
  me: IUser
  permissions: CommentPermissions
}> = ({
  columns,
  currentExplore,
  currentModel,
  loadingExplore,
  comments,
  addComment,
  editComment,
  deleteComment,
  authors,
  me,
  permissions,
}) => {
  const [search, setSearch] = useState('')
  const [shownColumns, setShownColumns] = useState([
    ...columns
      .filter((d) => {
        return d.default
      })
      .map((d) => d.rowValueDescriptor),
  ])
  const [hasDescription, setHasDescription] = useState([])
  const [hasTags, setHasTags] = useState([])
  const [hasComments, setHasComments] = useState([])
  const [fieldTypes, setFieldTypes] = useState([])
  const [selectedFields, setSelectedFields] = useState([])

  const typeMaker = (field: ILookmlModelExploreField) => {
    return humanize(field.type.split('_')[0])
  }

  if (loadingExplore) {
    return (
      <Main p="xxlarge">
        <Flex alignItems="center" height="100%" justifyContent="center">
          <Spinner />
        </Flex>
      </Main>
    )
  }

  if (
    currentModel &&
    currentExplore &&
    currentModel.name === currentExplore.model_name
  ) {
    const fields = flatten(values(currentExplore.fields))
      .map((f) => typeMaker(f))
      .filter((value, index, self) => self.indexOf(value) === index)

    const revivedComments = JSON.parse(comments)
    const encExplores = Object.keys(revivedComments)
    const commentObj = encExplores.includes(currentExplore.name)
      ? revivedComments[currentExplore.name]
      : (revivedComments[currentExplore.name] = {})

    const allFilters = hasDescription.concat(
      hasTags,
      hasComments,
      fieldTypes,
      selectedFields
    )
    const groups = orderBy(
      toPairs(
        groupBy(
          flatten(values(currentExplore.fields)).filter((f) => {
            const commentFlag = !!(
              commentObj[f.name] && commentObj[f.name].length > 0
            )
            return (
              !f.hidden &&
              (allFilters.length === 0 ||
                ((hasDescription.length === 0 ||
                  (hasDescription.includes('yes') && f.description) ||
                  (hasDescription.includes('no') && !f.description)) &&
                  (hasTags.length === 0 ||
                    (hasTags.includes('yes') && f.tags.length > 0) ||
                    (hasTags.includes('no') && f.tags.length === 0)) &&
                  (hasComments.length === 0 ||
                    (hasComments.includes('yes') && commentFlag) ||
                    (hasComments.includes('no') && !commentFlag)) &&
                  (fieldTypes.length === 0 ||
                    (fieldTypes.includes('dimensions') &&
                      f.category === DIMENSION) ||
                    (fieldTypes.includes('measures') &&
                      f.category === MEASURE)) &&
                  (selectedFields.length === 0 ||
                    selectedFields.includes(typeMaker(f)))))
            )
          }),
          (f) => f.view_label
        )
      ),
      ([group]) => group
    )

    return (
      <Main pt="large">
        <Flex
          flexDirection="row"
          justifyContent="space-between"
          mt="large"
          mb="xxlarge"
          pl="xxlarge"
          pr="xxlarge"
        >
          <FlexItem>
            <Heading as="h1" fontWeight="semiBold">
              {currentModel.label}
            </Heading>
            <Heading as="h4" color="text1">
              Select a field for more information.
            </Heading>
          </FlexItem>
          <FlexItem>
            <ExternalLink target="_blank" href={exploreURL(currentExplore)}>
              <ButtonOutline mr="small">Explore</ButtonOutline>
            </ExternalLink>
            <ViewOptions
              columns={columns.filter((d) => {
                return d.rowValueDescriptor !== 'comment'
              })}
              shownColumns={shownColumns}
              setShownColumns={setShownColumns}
            />
          </FlexItem>
        </Flex>
        <Flex mt="xlarge" pl="xxlarge" pr="xxlarge">
          <FlexItem width="350px">
            <ExploreSearch
              hideSearchIcon
              placeholder="Filter fields in this Explore"
              mt="medium"
              onChange={setSearch}
              value={search}
            />
          </FlexItem>
        </Flex>

        <QuickSearch
          selectedFields={selectedFields}
          fields={fields}
          fieldTypes={fieldTypes}
          hasDescription={hasDescription}
          hasTags={hasTags}
          hasComments={hasComments}
          setSelectedFields={setSelectedFields}
          setFieldTypes={setFieldTypes}
          setHasDescription={setHasDescription}
          setHasTags={setHasTags}
          setHasComments={setHasComments}
          showComments={!permissions.disabled}
        />

        <Box>
          {groups.map((group) => {
            if (group[1].length > 0) {
              return (
                <Fields
                  columns={columns}
                  explore={currentExplore}
                  fields={group[1]}
                  key={group[0]}
                  label={group[0]}
                  model={currentModel}
                  search={search}
                  shownColumns={shownColumns}
                  comments={comments}
                  addComment={addComment}
                  editComment={editComment}
                  deleteComment={deleteComment}
                  authors={authors}
                  me={me}
                  permissions={permissions}
                />
              )
            }
            return undefined
          })}
        </Box>
      </Main>
    )
  } else {
    return (
      <FullPage>
        <div style={{ width: '30%' }}>
          <img
            src={
              'https://marketplace-api.looker.com/app-assets/data_dictionary_2x.png'
            }
            alt="Empty Image"
          />
        </div>
        <IntroText>
          Click on one of the Explores to the left to begin searching through
          your data. Youâ€™ll see labels, descriptions, SQL definitions, and more
          for each field.
        </IntroText>
      </FullPage>
    )
  }
}
