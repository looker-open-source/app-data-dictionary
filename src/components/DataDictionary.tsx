/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { PureComponent, useState } from "react";
import {
  Chip,
  Flex,
  FlexItem,
  Heading,
  Spinner,
  theme,
} from "@looker/components";
import humanize from 'humanize-string'
import styled, { ThemeProvider } from "styled-components";
import { useAllModels } from "../utils/fetchers";
import "./styles.css";
import { PanelFields } from "./PanelFields";
import SidebarToggle from "./SidebarToggle";
import { useCurrentModel, useCurrentExplore } from "../utils/routes"
import { ColumnDescriptor } from "./interfaces";
import { SQLSnippet } from "./SQLSnippet";
import { Sidebar } from './Sidebar'
import { SidebarStyleProps } from "./interfaces";
import { NoModelsAvailable } from "./NoModelsAvailable";
import { ILookmlModelExploreField } from "@looker/sdk";
import { CategorizedLabel } from './CategorizedLabel'

export const columns: ColumnDescriptor[] = [
  {
    name: 'field-label',
    label: 'Field Label',
    rowValueDescriptor: 'label_short',
    formatter: (x: any, isRow: boolean, field: ILookmlModelExploreField) => {
      if (field.field_group_label && field.field_group_variant) {
        return `${field.field_group_label} ${field.field_group_variant}`
      } else {
        return x
      }
    },
    minWidth: '12em',
  },
  {
    name: 'category',
    label: 'Category',
    rowValueDescriptor: 'category',
    formatter: (x: any, isRow: boolean, field: ILookmlModelExploreField) => {
      return <CategorizedLabel label={x} category={field.category} />
    },
    minWidth: '12em',
  },
  {
    name: 'description',
    label: 'Description',
    rowValueDescriptor: 'description',
    formatter: (x: any, isRow: boolean) => {
      if (x && isRow && x.length > 200) {
        return x.substring(0, 200) + '...'
      }
      return x
    },
    maxWidth: '20em',
  },
  {
    name: 'lookml-name',
    label: 'LookML Name',
    rowValueDescriptor: 'name',
    formatter: (x: any) => {
      return x.replace(/\./g, '.\u200B');
    },
    minWidth: '15em',
  },
  {
    name: 'type',
    label: 'Type',
    rowValueDescriptor: 'type',
    formatter: (x: any) => humanize(x),
    minWidth: '8em',
  },
  {
    label: 'SQL',
    rowValueDescriptor: 'sql',
    formatter: (x: any, isRow: boolean) => {
      return (<SQLSnippet isRow={isRow} src={x} />)
    },
    maxWidth: '20em',
    name: 'sql',
  },
  {
    name: 'tags',
    label: 'Tags',
    rowValueDescriptor: 'tags',
    formatter: (tags: any) => {
      return tags.map((tag: string) => (
        <Chip disabled>{tag}</Chip>
      ))
    },
  },
]

export const DataDictionary: React.FC<{}> = () => {
  const unfilteredModels = useAllModels()
  const currentModel = useCurrentModel()
  const [sidebarOpen, setSidebarOpen] = React.useState(true)
  const [search, setSearch] = React.useState('')
  const { currentExplore, loadingExplore } = useCurrentExplore()

  let models

  if (unfilteredModels) {
    models = unfilteredModels.filter(m => m.explores && m.explores.some(e => !e.hidden))
    if (!models.length) {
      return <NoModelsAvailable />
    }
  }

  if (!models) {
    return <Flex alignItems="center" height="100%" justifyContent="center"><Spinner /></Flex>
  }

  const toggleFn = () => setSidebarOpen(!sidebarOpen);

  return (
    <ThemeProvider theme={theme}>
      <div style={{minWidth: "1200px"}}>
        <PageHeader>
          <FlexItem>
            <Heading as="h1" fontSize="xlarge" fontWeight="semiBold" mb="xsmall">
              Data Dictionary
            </Heading>
          </FlexItem>
        </PageHeader>
        <PageLayout open={sidebarOpen}>
          <LayoutSidebar>
            {sidebarOpen && <Sidebar
              currentExplore={currentExplore}
              currentModel={currentModel}
              loadingExplore={loadingExplore}
              models={models}
              search={search}
              setSearch={setSearch}
            />}
          </LayoutSidebar>
          <SidebarDivider open={sidebarOpen}>
            <SidebarToggle
              isOpen={sidebarOpen}
              onClick={toggleFn}
              headerHeight="114px"
            />
          </SidebarDivider>
          <PageContent>
            <PanelFields
              columns={columns}
              currentExplore={currentExplore}
              currentModel={currentModel}
              loadingExplore={loadingExplore}
              model={currentModel}
            />
          </PageContent>
        </PageLayout>
      </div>
    </ThemeProvider>
  );
}


const PageHeader = styled(Flex)`
  background-color: ${theme.colors.palette.purple400};
  background-position: 100% 0;
  background-repeat: no-repeat;
  background-size: 836px 120px;
  padding: ${theme.space.large};
  background-image: url('https://marketplace-api.looker.com/misc/INC-100373/spirals.png');

  h1 {
    color: ${theme.colors.palette.white};
  }
`;

const PageLayout = styled.div<SidebarStyleProps>`
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: ${({ open }) =>
    open ? "16.625rem 0 1fr" : "1.5rem 0 1fr"};
  grid-template-areas: "sidebar divider main";
  position: relative;
`;

const PageContent = styled.div`
  grid-area: main;
  position: relative;
`;

const LayoutSidebar = styled.aside`
  position: absolute;
  top: 0;
  bottom: 0;
  width: 16.625rem;
  grid-area: sidebar;
  z-index: 0;
`;

const SidebarDivider = styled.div<SidebarStyleProps>`
  transition: border 0.3s;
  border-left: 1px solid
    ${({ theme, open }) =>
      open ? theme.colors.palette.charcoal200 : "transparent"};
  grid-area: divider;
  overflow: visible;
  position: relative;
  &:hover {
    border-left: 1px solid
      ${({ theme, open }) =>
        open
          ? theme.colors.palette.charcoal300
          : theme.colors.palette.charcoal200};
  }
`;
