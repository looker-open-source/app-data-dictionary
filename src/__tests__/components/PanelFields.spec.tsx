/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import { screen } from '@testing-library/react'
import { PanelFields } from '../../components/PanelFields'
import { renderWithExtensionContext } from '../test_utils/render_with_extension'
import { columns } from '../../components/DataDictionary'
import { mockCurrentModel, mockCurrentExplore } from '../MockData/MockData'

jest.mock('../../components/DetailDrawer', () => ({
  DetailDrawer: () => 'DetailDrawer',
}))

jest.mock('../../components/Fields', () => ({
  Fields: () => 'Fields',
}))

jest.mock('../../components/ViewOptions', () => ({
  ViewOptions: () => 'ViewOptions',
}))

jest.mock('../../components/ExternalLink', () => ({
  ExternalLink: () => 'ExternalLink',
}))

jest.mock('../../components/QuickSearch', () => ({
  QuickSearch: () => 'QuickSearch',
}))

jest.mock('../../utils/routes', () => {
  return {
    useCurrentModel: jest.fn(() => {
      return mockCurrentModel
    }),
    useCurrentExplore: jest.fn(() => {
      return mockCurrentExplore
    }),
  }
})

describe('<PanelFields>', () => {
  it('renders correctly', () => {
    renderWithExtensionContext(
      <PanelFields
        currentModel={mockCurrentModel}
        currentExplore={mockCurrentExplore}
        loadingExplore={null}
        columns={columns}
        comments={'{}'}
        addComment={jest.fn()}
        editComment={jest.fn()}
        deleteComment={jest.fn()}
        authors={[]}
        me={{}}
        permissions={{}}
      />
    )
    expect(screen.getByText('Snowflake Usage')).toBeInTheDocument()
    expect(
      screen.getByText('Select a field for more information.')
    ).toBeInTheDocument()
    expect(screen.getByText(/ExternalLink/)).toBeInTheDocument()
    expect(screen.getByText(/ViewOptions/)).toBeInTheDocument()
    expect(screen.getByText(/Fields/)).toBeInTheDocument()
  })
})
