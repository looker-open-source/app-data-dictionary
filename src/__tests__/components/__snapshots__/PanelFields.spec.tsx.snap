// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Box-sc-xja7ec sc-EHOje bCXiAY"
>
  <div
    className="Flex-sc-vm49hu ljMeNV"
  >
    <div
      className="FlexItem-sc-1re6mwg eLXSKG"
    >
      <h1
        className="TextBase-sc-1rkc9wq Heading-sc-cg3kf8 gbsXsz"
        fontSize="xxlarge"
        fontWeight="semiBold"
      >
        Snowflake Usage
      </h1>
      <h4
        className="TextBase-sc-1rkc9wq Heading-sc-cg3kf8 fLuWuF"
        fontSize="medium"
        fontWeight="normal"
      >
        Select a field for more information.
      </h4>
    </div>
    <div
      className="FlexItem-sc-1re6mwg eLXSKG"
    >
      <a
        className="Link-sc-9w9yc8 hlbwPV"
        color="palette.blue500"
        href="/explore/snowflake_usage/query_history"
        onClick={[Function]}
        target="_blank"
        textDecoration="none"
      >
        <button
          className="ButtonOuter-sc-wugisi cHalYv ButtonBase-sc-18wmtoc ButtonOutline-sc-275zsb jhloJm"
          onBlur={[Function]}
          onKeyUp={[Function]}
          size="medium"
        >
          Explore
        </button>
      </a>
      ViewOptions
    </div>
  </div>
  <div
    className="Flex-sc-vm49hu eoGueT"
  >
    <div
      className="FlexItem-sc-1re6mwg dNebzy"
      width="350px"
    >
      <div
        className="Flex-sc-vm49hu eOMeYs InputSearch-sc-2764tg sc-htoDjs knjixi"
        fontSize="small"
        onMouseDown={[Function]}
      >
        <input
          className="StyledInput-sc-f13pa jzRlTN InputText-sc-vhjogy dQduGO"
          onChange={[Function]}
          placeholder="Filter fields in this Explore"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="Box-sc-xja7ec sc-bdVaJa cOZUhj"
  >
    <div
      className="Flex-sc-vm49hu flLYsP"
    >
      <div
        className="FlexItem-sc-1re6mwg sc-htpNat kVstQD"
      >
        <h6
          className="TextBase-sc-1rkc9wq Heading-sc-cg3kf8 sc-bwzfXH jKtyhP"
          fontSize="xsmall"
          fontWeight="normal"
        >
          Has Description
        </h6>
        <div
          className="Box-sc-xja7ec gqzGoP ButtonGroup-sc-chg3ap kiShRM"
          display="inline-flex"
          fontSize="small"
        >
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="yes"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Yes
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="no"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              No
            </span>
          </label>
        </div>
      </div>
      <div
        className="FlexItem-sc-1re6mwg sc-htpNat kVstQD"
      >
        <h6
          className="TextBase-sc-1rkc9wq Heading-sc-cg3kf8 sc-bwzfXH jKtyhP"
          fontSize="xsmall"
          fontWeight="normal"
        >
          Fields
        </h6>
        <div
          className="Box-sc-xja7ec gqzGoP ButtonGroup-sc-chg3ap kiShRM"
          display="inline-flex"
          fontSize="small"
        >
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="dimensions"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Dimension
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="measures"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Measure
            </span>
          </label>
        </div>
      </div>
      <div
        className="FlexItem-sc-1re6mwg sc-htpNat kVstQD"
      >
        <h6
          className="TextBase-sc-1rkc9wq Heading-sc-cg3kf8 sc-bwzfXH jKtyhP"
          fontSize="xsmall"
          fontWeight="normal"
        >
          Has Tags
        </h6>
        <div
          className="Box-sc-xja7ec gqzGoP ButtonGroup-sc-chg3ap kiShRM"
          display="inline-flex"
          fontSize="small"
        >
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="yes"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Yes
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="no"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              No
            </span>
          </label>
        </div>
      </div>
      <div
        className="FlexItem-sc-1re6mwg sc-htpNat kVstQD"
      >
        <h6
          className="TextBase-sc-1rkc9wq Heading-sc-cg3kf8 sc-bwzfXH jKtyhP"
          fontSize="xsmall"
          fontWeight="normal"
        >
          Type
        </h6>
        <div
          className="Box-sc-xja7ec gqzGoP ButtonGroup-sc-chg3ap kiShRM"
          display="inline-flex"
          fontSize="small"
        >
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="String"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              String
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="Date"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Date
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="Number"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Number
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="Yesno"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Yesno
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="Count"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Count
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="Average"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Average
            </span>
          </label>
          <label
            className="ButtonItemLabel-sc-1beway8 iqRTth ButtonItem-sc-11a1q7g NkLis"
            fontFamily="brand"
            selected={false}
            type="checkbox"
          >
            <input
              checked={false}
              className="ButtonInput-sc-1346jqs cOfiUU"
              onChange={[Function]}
              type="checkbox"
              value="Sum"
            />
            <span
              className="ButtonItemLabelText-sc-1xqs192 jSfzmm"
            >
              Sum
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  Box
</div>
`;
